newIDE/app/src/Profile/Achievement/AchievementList.js

```diff
@@ -0,0 +1,90 @@
+// @flow
+import React, { useEffect, useState } from 'react';
+import { DateFormat } from '@lingui/macro';
+
+import { Column, Line } from '../../UI/Grid';
+import Text from '../../UI/Text';
+import {
+  type Badge,
+  type Achievement,
+} from '../../Utils/GDevelopServices/User';
+
+type Props = {|
+  badges: Array<Badge>,
+  achievements: Array<Achievement>,
+|};
+
+const achievementBaseline = {
+  width: '100%',
+  padding: '0 20px',
+};
+
+const styles = {
+  achievementsContainer: {
+    maxHeight: 250,
+    overflowY: 'scroll',
+  },
+  unlockedAchievement: {
+    ...achievementBaseline,
+    color: 'green',
+    display: 'flex',
+    justifyContent: 'space-between',
+  },
+  lockedAchievement: {
+    ...achievementBaseline,
+    color: '#CCCCCC',
+  },
+};
+
+const AchievementList = ({ badges, achievements }: Props) => {
+  const [formattedAchievements, setFormattedAchievements] = useState([]);
+  useEffect(
+    () => {
+      const badgeByAchievementId = badges.reduce((acc, badge) => {
+        acc[badge.achievementId] = badge;
+        return acc;
+      }, {});
+
+      setFormattedAchievements(
+        achievements.map(achievement => {
+          if (Object.keys(badgeByAchievementId).includes(achievement.id)) {
```

Comment:
You can probably just do
```suggestion
          if (!!badgeByAchievementId[achievement.id]) {
```

if the key is not defined or the value undefined, it will be falsy.
This avoids to do a O(n) search in a temporary array for each achievement.